server/mergin/sync/public_api_v2_controller.py

```diff
@@ -43,3 +44,32 @@ def delete_project_now(id_):
     db.session.commit()
 
     return NoContent, 204
+
+
+@auth_required
+def update_project(id_):
+    """Rename project
+
+    :param id_: Project_id
+    :type id_: str
+    """
+    project = require_project_by_uuid(id_, ProjectPermissions.Update)
+    new_name = request.json["name"]
+
+    if not is_name_allowed(new_name):
+        return (
+            jsonify(
+                code="InvalidProjectName", detail="Entered project name is invalid"
+            ),
+            400,
+        )
+    new_name_exists = Project.query.filter_by(
+        workspace_id=project.workspace_id, name=new_name
```

Comment:
@luxusko I think we should filter by `name=new_name.strip()`, right? 

because if there is a trailing space in new_name, it is not catched by is_name_allowed and here we should check the name which we want to write to db (there could be already a different project without the whitespace)

I didn't notice you moved the .strip() to the end.